### ISSUE TITLE
[BUG/ISSUE] TOMAS40 simulation dies with negative O3 concentration

### ISSUE THREAD

**Issue by yantosca** [geoschem/geos-chem#1101]
### What institution are you from?
GCST

### Description of the problem
An integration test of 13.4.0-alpha.17 (GEOS-Chem Classic) revealed this error:

```console
AEROPHYS: This run uses coupled condensation-nucleation scheme with pseudo-steady state H2SO4
   Nucleation: Ternary (Napari et al. 2002) and Binary (Vehkamaki et al. 2002)
 Debug TOMAS: H2SO4RATE =   9.9633690471968239E-002
 gas production rate too high in getH2SO4conc
   6.7858064105479663        1.1848456040614934E-003   5722.7368859910348        4.4747283363661822E-003   7.7203964456297458E-004
---> DATE: 2019/07/01  UTC: 00:30  X-HRS:      0.500000
 WARNING: Negative concentration for species O3 at (I,J,L) =           39          37           1
===============================================================================
GEOS-Chem ERROR:
 -> at DO_TEND (in module GeosCore/mixing_mod.F90)
===============================================================================

===============================================================================
GEOS-Chem ERROR: Error encountred in "DO_TEND"!
 -> at DO_MIXING (in module GeosCore/mixing_mod.F90)
===============================================================================

===============================================================================
GEOS-CHEM ERROR: Error encountered in "Do_Mixing"!
STOP at  -> at GEOS-Chem (in GeosCore/main.F90)
===============================================================================
```
It appears that a negative O3 concentration is happening.  This perhaps is due to the updated sulfur chemistry via KPP.  Perhaps the simplest fix is to reset negative concentrations to zero somewhere in TOMAS.  @theloniuspunk

**Comment by yantosca** [comment]
Ah OK, that might be it.  I can change the default behavior in the run directory script if that is not the case.

**Comment by yantosca** [comment]
Actually it is already set to false by default.
```console
------------------------+------------------------------------------------------
%%% CONVECTION MENU %%% :
Turn on Cloud Conv?     : T
Turn on PBL Mixing?     : T
 => Use non-local PBL?  : F
```
Back to square one.

**Comment by theloniuspunk** [comment]
This is the same issue: https://github.com/geoschem/geos-chem/issues/1025#issuecomment-1009380752 There wasn't any follow up after your suggestion to move to v13.3.3.

**Comment by theloniuspunk** [comment]
Right, the issue is that setting non-local PBL mixing to F causes the issue and that TOMAS requires non-local PBL mixing to be F. Do you have any idea as to why mixing evenly through the PBL (i.e. non-local PBL mixing = F) would cause this O3 error?

**Comment by yantosca** [comment]
I don't have a great idea at this moment @theloniuspunk.  We are going to freeze 13.4.0 this week but perhaps a fix could go into a patch version such as 13.4.1.

**Comment by theloniuspunk** [comment]
I just sent a note to Betty Croft (she's not on here right?). I feel like she found a fix (it's possible that she just switched back to an older version). I'll let you know if she knows the fix.

**Comment by theloniuspunk** [comment]
From Betty Croft: 

In my experience, the negative ozone issue occurred in GCClassic (v13.2.1) - but not in GCHP (v13.3.4). I got GCC v13.2.1 to run by resetting the negative to a small positive in mixing_mod.F90 and preventing the failure condition. I printed out whenever the correction was made and it was very infrequent (mostly just in one grid box in Germany region) and the need for a correction seemed to be less after the model ran. This issue did not seem to be affecting my Arctic simulations for MOSAiC - so I left it there for now. 

I am not sure that it is a satisfactory fix because I still do not understand why the problem occurred. Note that my issue with GCC v13.2.1 was with TOMAS15 and non-local PBL to false (which TOMAS needs to run). Below is what I changed in mixing_mod.F90.

! Check for negative concentrations                                                                                
       ! KLUDGE: skip the warning message for CH4_SAB, which can be                                                                   
       ! negative (it's a soil absorption flux). The TagCH4 simulation                                                                 
       ! is not used regularly as of Feb 2021 -- fix this later if                                                                    
       ! need by. (bmy, 2/25/21)                                                                                     
       IF ( State_Chm%Species(I,J,L,N) < 0.0_fp ) THEN
!bc - look for negative and reset to small positive                                                                              
         print *, 'Found negative ', N, State_Chm%Species(I,J,L,N)
         State_Chm%Species(I,J,L,N)=1.e-26_fp
        IF ( N /= id_CH4_SAB ) THEN
         Print*, 'WARNING: Negative concentration for species ',   &
             TRIM( SpcInfo%Name), ' at (I,J,L) = ', I, J, L
         ErrorMsg = 'Negative species concentations encountered.' // &
              ' This may be fixed by increasing the'    // &
              ' background concentration or by shortening' // &
              ' the transport time step.'
         RC = GC_FAILURE
        ENDIF
       ENDIF

I think I commented out the failure too.

**Comment by yantosca** [comment]
Thanks @theloniuspunk.  I can add something like this for TOMAS.  Not sure I would want to do that in other simulations as of yet, as negative concentrations should generate a failure in e.g. the regular fullchem model.

Also, do you know if Betty is on Github and and if so, what her handle is?

**Comment by theloniuspunk** [comment]
Thanks! I don't think she is on here, I couldn't find her at least.

**Comment by yantosca** [comment]
I've added Betty's fix in commit 4885456f497749dc5ea3e27abc3ac41a33d3e1b0.  For now I've added this in an `#ifdef TOMAS` block so that the error trap will still work for other simulations as before.  This will go into 13.4.0.  We can further investigate this issue and issue a patch version if a solution for the negative values is found.

**Comment by theloniuspunk** [comment]
Thanks, Bob!

**Comment by yantosca** [comment]
Just confirmed that this works as advertised:

```console:
...
 AEROPHYS: This run uses coupled condensation-nucleation scheme with pseudo-steady state H2SO4
   Nucleation: Ternary (Napari et al. 2002) and Binary (Vehkamaki et al. 2002)
 Debug TOMAS: H2SO4RATE =   9.9633690471968239E-002
 gas production rate too high in getH2SO4conc
   6.7858064105479663        1.1848456040614934E-003   5722.7368859910348        4.4747283363661822E-003   7.7203964456297458E-004
---> DATE: 2019/07/01  UTC: 00:30  X-HRS:      0.500000
*Found negative          180  -3.1127737922001793E-008
     - Creating file for SpeciesConc; reference = 20190701 000000
        with filename = OutputDir/GEOSChem.SpeciesConc.20190701_0000z.nc4
     - Creating file for Metrics; reference = 20190701 000000
        with filename = OutputDir/GEOSChem.Metrics.20190701_0000z.nc4
     - Creating file for Restart; reference = 20190701 010000
        with filename = ./GEOSChem.Restart.20190701_0100z.nc4
---> DATE: 2019/07/01  UTC: 01:00  X-HRS:      1.000000
     - INITIALIZE: Diag arrays zeroed!
     - INITIALIZE: Diag counters zeroed!
     - CLEANUP: deallocating arrays now...

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%     To compute the OH metrics, execute the Python     %%%%%
%%%%%       script "metrics.py" in this run directory.      %%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

===> SIMULATION END TIME: 2022/01/21 14:20 <===
```

**Comment by yantosca** [comment]
This has now been merged into the 13.4.0 development stream, so we can close this issue.
